PurchasesCoreSwift/Identity/PurchaserInfoManager.swift

```diff
@@ -0,0 +1,191 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchaserInfoManager.swift
+//
+//  Created by Joshua Liebowitz on 8/5/21.
+
+import Foundation
+
+public typealias ReceivePurchaserInfoBlock = (PurchaserInfo?, Error?) -> Void
+
+@objc(RCPurchaserInfoManagerDelegate) public protocol PurchaserInfoManagerDelegate: NSObjectProtocol {
+
+    @objc(purchaserInfoManagerDidReceiveUpdatedPurchaserInfo:)
+    func purchaserInfoManagerDidReceiveUpdated(purchaserInfo: PurchaserInfo)
+
+}
+
+// TODO (post-migration) make all the things internal, including the protocol.
+@objc(RCPurchaserInfoManager) public class PurchaserInfoManager: NSObject {
+
+    @objc public weak var delegate: PurchaserInfoManagerDelegate?
+
+    private(set) var lastSentPurchaserInfo: PurchaserInfo?
+    private let operationDispatcher: OperationDispatcher
+    private let deviceCache: DeviceCache
+    private let backend: Backend
+    private let systemInfo: SystemInfo
+    private let purchaserInfoCacheLock = NSRecursiveLock()
+
+    @objc public init(operationDispatcher: OperationDispatcher,
+                      deviceCache: DeviceCache,
+                      backend: Backend,
+                      systemInfo: SystemInfo) {
+        self.operationDispatcher = operationDispatcher
+        self.deviceCache = deviceCache
+        self.backend = backend
+        self.systemInfo = systemInfo
+    }
+
+    @objc public func fetchAndCachePurchaserInfo(appUserID: String,
+                                                 isAppBackgrounded: Bool,
+                                                 completion: ReceivePurchaserInfoBlock?) {
+        deviceCache.setPurchaserInfoCacheTimestampToNow(appUserID: appUserID)
```

Comment:
probably not part of the migration, but setting the cache timestamp feels like something that should be in the `cache` method itself?

Reply:
yeah, so, about the purpose of this: it's for concurrency control. 
The idea is that if you ask for purchaser info again while the request is in flight, we'll look at the cache date, it'll be very recent, and we won't re-fetch. 

This could be refactored into a cleaner system, but for the sake of migrating while minimizing changes, we might not want to do it right now. 

What we _could_ do is perhaps make it more explicit? i.e.: 

```swift
private func setCacheTimestampToNowToPreventConcurrentPurchaserInfoUpdates() {
    deviceCache.setPurchaserInfoCacheTimestampToNow(appUserID: appUserID)
}

@objc public func fetchAndCachePurchaserInfo(appUserID: String,
                                                  isAppBackgrounded: Bool,
                                                  completion: ReceivePurchaserInfoBlock?) {
    setCacheTimestampToNowToPreventConcurrentPurchaserInfoUpdates()

```

Reply:
It definitely does. Opening issue for investigation.

Reply:
https://github.com/RevenueCat/purchases-ios/issues/723

Reply:
thoughts on doing a wrapper or at least a comment for it in this PR?

Reply:
I think updating the func naming would be helpful, but maybe we should consider refactoring this in the future.

Reply:
On it.